services/blockchain-coordinator/shared/scheduler.js

```diff
@@ -63,13 +63,35 @@ const getInProgressJobCount = async (queue) => {
 };
 
 const getLiveIndexingJobCount = async () => {
-	const messageQueueJobCount = await getInProgressJobCount(blockIndexQueue);
+	const messageQueueJobCount = await getInProgressJobCount(blockMessageQueue);
 	const indexerLiveIndexingJobCount = await getLiveIndexingJobCountFromIndexer();
 	return messageQueueJobCount + indexerLiveIndexingJobCount;
 };
 
 const getRegisteredModuleAssets = () => registeredLiskModules;
 
+const waitForJobCountToFallBelowThreshold = (resolve) => new Promise((res) => {
+	if (!resolve) resolve = res;
+	if (intervalID) {
+		clearInterval(intervalID);
+		intervalID = null;
+	}
+
+	return getLiveIndexingJobCount()
+		.then((count) => {
+			const { skipThreshold } = config.job.indexMissingBlocks;
+			return count < skipThreshold
+				? resolve(true)
+				: (() => {
+					logger.info(`In progress job count (${String(count).padStart(5, ' ')}) not yet below the threshold (${skipThreshold}). Waiting for ${REFRESH_INTERVAL}ms to re-check the job count before scheduling the next batch.`);
+					intervalID = setInterval(
+						waitForJobCountToFallBelowThreshold.bind(null, resolve),
+						REFRESH_INTERVAL,
+					);
+				})();
+		});
+});
+
```

Comment:
would it be better to simplify the implementation by using delay function like this? https://github.com/LiskHQ/lisk-service/blob/51b074adc43117d70d514fc622f4ce7b8d09c402/services/blockchain-connector/shared/utils/delay.js#L16

```suggestion
const waitForJobCountToFallBelowThreshold = async () => {
	while (true) {
		const count = await getLiveIndexingJobCount();
		const { skipThreshold } = config.job.indexMissingBlocks;

		if (count < skipThreshold) return;

		logger.info(`In progress job count (${String(count).padStart(5, ' ')}) not yet below the threshold (${skipThreshold}). Waiting for ${REFRESH_INTERVAL}ms to re-check the job count before scheduling the next batch.`);
		await delay(REFRESH_INTERVAL);
	}
};

```